docs/ref/templates/builtins.txt

```diff
@@ -959,66 +959,100 @@ output (as a string) inside a variable. This is useful if you want to use
 
 .. versionadded:: 5.1
 
-Outputs the query string from a given :class:`~django.http.QueryDict` instance,
-if provided, or ``request.GET`` if not and the
-``django.template.context_processors.request`` context processor is enabled.
-If the ``QueryDict`` is empty, then the output will be an empty string.
-Otherwise, the query string will be returned with a leading ``"?"``.
+Outputs a URL-encoded formatted query string based on the provided parameters.
 
-If not using the ``django.template.context_processors.request`` context
-processor, you must pass either the ``request`` into the template context or a
-``QueryDict`` instance into this tag.
+This tag requires a :class:`~django.http.QueryDict` instance, which defaults to
+:attr:`request.GET <django.http.HttpRequest.GET>` if none is provided, to
+generate the resulting query string from the passed key-value pairs.
 
-The following example outputs the current query string verbatim. So if the
-query string is ``?color=green&size=M``, the output would be
-``?color=green&size=M``:
+If the :class:`~django.http.QueryDict` is empty and no additional parameters
+are provided, an empty string is returned. A non-empty result includes a
+leading ``"?"``.
+
+.. admonition:: Using ``request.GET`` as default
+
+    To use ``request.GET`` as the default ``QueryDict`` instance, the
+    ``django.template.context_processors.request`` context processor should be
+    enabled. If it's not enabled, you must either explicitly pass the
+    ``request`` object into the template context, or provide a ``QueryDict``
+    instance to this tag.
+
+Basic usage
+~~~~~~~~~~~
 
 .. code-block:: html+django
 
     {% querystring %}
 
-You can also pass in a custom ``QueryDict`` that will be used instead of
-``request.GET``:
+Outputs the current query string verbatim. So if the query string is
+``?color=green``, the output would be ``?color=green``.
+
+.. code-block:: html+django
+
+    {% querystring size="M" %}
+
+Outputs the current query string with the addition of the ``size`` parameter.
+Following the previous example, the output would be ``?color=green&size=M``.
+
+Custom QueryDict
+~~~~~~~~~~~~~~~~
 
 .. code-block:: html+django
 
     {% querystring my_query_dict %}
 
-Each keyword argument will be added to the query string, replacing any existing
-value for that key. With the query string ``?color=blue``, the following would
-result in ``?color=red&size=S``:
+You can provide a custom ``QueryDict`` to be used instead of ``request.GET``.
+So if ``my_query_dict`` is ``<QueryDict: {'color': ['green']}>``, this outputs
+``?color=green``.
```

Comment:
```suggestion
So if ``my_query_dict`` is ``<QueryDict: {'color': ['blue']}>``, this outputs
``?color=blue``.
```
I think given the previous examples say green is in the default QueryDict, it might be nice to update the color.

Also this used to say
> Each keyword argument will be added to the query string, replacing any existing
value for that key.

Perhaps we want to keep that sentence and update the example to something like `<QueryDict: {'color': ['blue'], 'fabric': ['silk']}>`

Reply:
I love your proposal of using a different color here (`blue`). I have made that modification, thank you for the suggestion.

Regarding the "keywork argument" comment, I think it's more appropriate in the following subsection (`Setting items`), and I have moved it there. Perhaps I should add a second example in that subsection updating a given `QueryDict`? :thinking:

Reply:
> Each keyword argument will be added to the query string, replacing any existing
value for that key.

If the current query string (from `request.GET`) was `?color=green` and the passed in query dict is `<QueryDict: {'size': ['M']}>` is the output `?color=green&size=M` or `?size=M` I wonder 🤔

Reply:
Is the latter, confirmed by looking at the implementation and IRL testing.

Then there is also this corner case which does not against the docs but I think it's a bit unexpected:

* query string from request is `?color=green`
* an explicit QueryDict is passed that is empty (but not None)
* `{% querystring empty_qs %}` will result in the empty string
* `{% querystring empty_qs something="foo" %}` will result in `?something=foo`

As mentioned, I think this aligns with the docs ("defaults to `request.GET` if none is provided.") but unsure if we should do something else about this case.